### ISSUE TITLE
info about the simiam

### ISSUE THREAD

**Issue by andre-rangel** [jdelacroix/simiam#10]
Hello there, I would like to more about the project and it's current status. Are there some plan to follow? Can we contribute? Why so many branches?

Thanks!

**Comment by jdelacroix** [comment]
Hello! You can read more about the project here: http://gritslab.gatech.edu/home/2013/10/sim-i-am. This mobile robot simulator is currently used for the programming assignments in the Coursera course, Control of Mobile Robots (https://www.coursera.org/course/conrob). Most of the branches are related to the course and the new robot, the QuickBot (http://o-botics.org/robots/quickbot/mooc/v1/). The course is ending soon, so I am preparing to merge the changes from the coursera-\* branches into the master branch soon. Afterwards, I will think about what features to add to the simulator next (higher fidelity physics and an upgrade to 3D are probably on top of my list).

This project is most definitely open for you and anybody else to contribute! Thank you for your questions and interest in the project!

**Comment by andre-rangel** [comment]
Great! I found this repository because I did the course and specially liked the simulator. Please, don't forget to post the new plans for the software. Upgrade to 3d would be awsome, make a platform to draw different obstacles, a record button to save the run in video and show in the course or send to friends. Anyway, would be great! Count me in.

**Comment by jdelacroix** [comment]
Those are great ideas! I think it would be useful to have a level editor for adding robots and obstacles to the environment graphically (rather than through the XML). Also, sharing recordings would have been an amazing features to have during the course.

Thanks!

**Comment by andre-rangel** [comment]
To improve the software, we may integrate the different parts of the simiam instead of typing in the matlab prompt to call other interfaces, as the level editor. This is an idea. Maybe we can make a especial version of the application for the course, with a submit button.
![simiam_flowchart](https://f.cloud.github.com/assets/6898578/2455795/ba602f7e-af07-11e3-9b89-898aea414498.jpg)

**Comment by jdelacroix** [comment]
I agree that the simulator (application) should be launched with a single command, and all of its functionality can be accessed from the start (rather than relying on launch, submit, edit, or other scripts). What do you think about integrating the level editor and simulator, so that a user can edit the world dynamically even if the simulation is already running? 

Another idea (if simulator and level editor are integrated): a user has a real robot and is using it to map the world. The sensor data is plotted in the simulator, and the user can then add a virtual version of the obstacle based on the sensor data.

**Comment by andre-rangel** [comment]
I like it! I think more iteraction should be attractive to people. The user will choose the environment or make a new one, add obstacles, mark the goal if necessary and add robots. A good customized environment for simulation. A lot of work to do!

**Comment by andre-rangel** [comment]
I made drawings with pencil with some suggestions for the main window of the simulator  integrated with the level editor. 
![simiamtela1bw](https://f.cloud.github.com/assets/6898578/2489327/32d733c4-b15d-11e3-95de-c6f75c0f70ad.png)

(1) menu : Open, new, close, save, choose, Robots, exit;
     Help: Manual, FQA;
     About: web Site, about this version.
 (2) Tabs .
 (3) tools.
 (4) hide tools.
 (5) squares.
 (6) circles.
 (7) triangles.
 (8) line.
 (9) curve.
 (10) change cursor (hand cursor<-> arrow cursor).
(11) Robots.
(12) play.
(13) record.
(14) time.
(15) screen.
(16) space for mensages.

**Comment by andre-rangel** [comment]
tabs can help to edit several levels with one open window.

**Comment by jdelacroix** [comment]
Nice design! I like that it is clean, and the toolbox/toolsidebar is a great idea, especially if it can be hidden away easily.

I had always envisioned the simulator UI to be a crossover between a maps and a music app. For example, https://raw.githubusercontent.com/gnome-design-team/gnome-mockups/master/music/music-album-playback.png and https://wiki.gnome.org/Design/Apps/Maps?action=AttachFile&do=get&target=main-view.png.

The map interface is natural, because it is a robot driving through an environment (which like a map, has obstacles and spaces to drive), while the simulation controls are similar to music controls (stop, pause, record, rewind, forward, etc.) A combination of the two with the above toolbar would be fantastic, I think. (Perhaps, I'll sit down this week and make a mockup too!)

On a more technical note, the simulator is using this package for the UI: http://www.mathworks.com/matlabcentral/fileexchange/29472-gridlayout, which is a little inflexible and hasn't been update in a while. I'm considering this package instead: http://www.mathworks.com/matlabcentral/fileexchange/27758-gui-layout-toolbox, which should make it straightforward to implement your mockup, for example.

**Comment by andre-rangel** [comment]
Thanks, I still don't know how much one can modify matlab ui, because my background is c++/Qt framework which gives you free control over every part of the ui or even make new widgets for interfaces. This last package seems to be really what we are looking for regarding layouts.

I liked the examples that you showed me. Before everything, I think we need a complete historyboard for the new simulator and every feature to have that clear in mind. 

We can modify the environment when the user clicks in the record button, showing how much time it is been recording, for example, and a different environment for play the recording. We may need more packages for these features too.

**Comment by jdelacroix** [comment]
That sounds like the most logical way forward. Re: UIs in MATLAB, they're not as feature rich and flexible as Qt, and often require poking at the Java objects underneath. For example, all images in the current UI are embedded via HTML in a MATLAB button. These buttons by default have borders, which you can only remove/turn off by accessing the internal Java object corresponding to that button.

Here's a list of widgets that are available: http://www.mathworks.com/help/matlab/component-selection.html

**Comment by andre-rangel** [comment]
![simiamtela1](https://cloud.githubusercontent.com/assets/6898578/2552231/1d1c8180-b697-11e3-8cd4-b8dafd3630fa.png)
![simiamtela2](https://cloud.githubusercontent.com/assets/6898578/2552233/252e6118-b697-11e3-955d-936024a6ca22.png)
![simiamtela3](https://cloud.githubusercontent.com/assets/6898578/2552238/2950ff4e-b697-11e3-9c25-97482e742ad0.png)
![simiamtela4](https://cloud.githubusercontent.com/assets/6898578/2552240/2d992ea0-b697-11e3-84fa-68d2b295cd08.png)
![simiamtela5](https://cloud.githubusercontent.com/assets/6898578/2552243/31200990-b697-11e3-8645-9c6334d0cfa5.png)
![simiamtela6](https://cloud.githubusercontent.com/assets/6898578/2552245/3cccb3d8-b697-11e3-957a-3229aa8674b0.png)
![simiamtela7](https://cloud.githubusercontent.com/assets/6898578/2552250/4242ed0a-b697-11e3-920a-231fce1c7601.png)
![simiamtela8](https://cloud.githubusercontent.com/assets/6898578/2552252/47c56104-b697-11e3-8cfa-6f2fe5688b32.png)
![simiamtela9](https://cloud.githubusercontent.com/assets/6898578/2552259/4e13549e-b697-11e3-925e-5f55d6bd23da.png)
![simiamtela10](https://cloud.githubusercontent.com/assets/6898578/2552261/5198c7c0-b697-11e3-93aa-194c84f902be.png)
![simiamtela11](https://cloud.githubusercontent.com/assets/6898578/2552263/56ea9a6e-b697-11e3-9115-5d8bf3a69abc.png)
![simiamtela12](https://cloud.githubusercontent.com/assets/6898578/2552268/5a9cc588-b697-11e3-9b25-744cce154b28.png)

**Comment by Dadwi** [comment]
Good morning , Is it now possible to add and to modify the environment architecture ? If yes how should I proceed .Furthermore can I add robots and change their behaviour .In which file should I do that ? Thank you
